airflow/kubernetes/pod_generator.py

```diff
@@ -460,9 +460,9 @@ def make_unique_pod_id(pod_id: str) -> str:
 
         safe_uuid = uuid.uuid4().hex  # safe uuid will always be less than 63 chars
         # Strip trailing '-' and '.' as they can't be followed by '.'
-        trimmed_pod_id = pod_id[:MAX_LABEL_LEN].rstrip('-.')
+        trimmed_pod_id = pod_id[:MAX_LABEL_LEN-len(safe_uuid)].rstrip('-.')
```

Comment:
Please also update the comment above to explain the logic. It's also probably better to split this into two lines

```suggestion
        label_prefix_length = MAX_LABEL_LEN - len(safe_uuid)
        trimmed_pod_id = pod_id[:label_prefix_length].rstrip('-.')
```

Because you probably won't like how Black formats this in a single line 🙂

Reply:
thanks. I have updated with these changes